onedal/linear_model/linear_model.cpp

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/linear_regression.hpp"
+
+#include "onedal/common.hpp"
+#include <regex>
+
+namespace py = pybind11;
+
+namespace oneapi::dal::python {
+
+namespace linear_model {
+
+template <typename Task, typename Ops>
+struct method2t {
+    method2t(const Task& task, const Ops& ops) : ops(ops) {}
+
+    template <typename Float>
+    auto operator()(const py::dict& params) {
+        using namespace dal::linear_regression;
+
+        const auto method = params["method"].cast<std::string>();
+        ONEDAL_PARAM_DISPATCH_VALUE(method, "norm_eq", ops, Float, method::norm_eq);
+        ONEDAL_PARAM_DISPATCH_VALUE(method, "by_default", ops, Float, method::norm_eq);
+        ONEDAL_PARAM_DISPATCH_THROW_INVALID_VALUE(method);
+    }
+
+    Ops ops;
+};
+
+auto get_onedal_result_options(const py::dict& params) {

```

Comment:
dont use `onedal` in the func name, inconsistent

Reply:
Nope. It is consistent. I.e. in [SVM](https://github.com/intel/scikit-learn-intelex/blob/e6495bbc43e5f84a147d29a54082567cc5901583/onedal/neighbors/neighbors.cpp#L103)